DEVELOPER_GUIDE.md

```diff
@@ -27,6 +30,50 @@ changes to Fluent Bit.
     - [Valgrind](#valgrind)
   - [Need more help?](#need-more-help)
 
+## Development Environment
+
+Development environments allow to work without any dependency locally.
+
+### Devcontainer
+
+A Development Container (or Dev Container for short) allows you to use a container as a full-featured development environment.
+
+Thanks to [Visual Studio Code integration](https://code.visualstudio.com/docs/devcontainers/containers), devcontainers are managed easily.
+
+#### Prerequirements
+- [docker](https://www.docker.com)
```

Comment:
Probably should be a container runtime, any should work

Reply:
I wish it supports any container runtime but I did not tried another one. [Its documantation says](https://code.visualstudio.com/docs/devcontainers/containers#_system-requirements) docker is required.

Reply:
We should probably reference the documentation directly I think, i.e. https://code.visualstudio.com/docs/devcontainers/containers#_system-requirements

The pre-reqs may change so if we just have a statement to say check there then we do not have to update the docs each time.